geospaas/nansat_ingestor/tests.py

```diff
@@ -396,3 +397,115 @@ def test_get_or_create_raises_AttributeError(self):
         # I am not sure which situations caused AttributeError, so this is not tested now (on 2019-02-15,
         # the S2 data access from the Norwegian ground segment was failing)
         pass
+
+
+class TestsForUpdateAbility(TestCase):
```

Comment:
You can inherit from BasetForTests and only redefine predefined_metadata_dict. The rest, except for test_for_examining_the_updating_purpose_of_ingestor_code can be removed.
```suggestion
class TestsForUpdateAbility(BasetForTests):
```

Reply:
We mock the whole test with that `@patch('os.path.isfile')`. In other words, this is the way of using the patching functionality. Otherwise we have to use it in another way (with context manager) which is not advantageous to implement one way or another. So, it can't be moved.